eq_translations/survey_schema.py

```diff
@@ -52,101 +41,74 @@ def translatable_items(self):
         :yield: A TranslatableItem
         :return: A generator
         """
+        for extractable_string in EXTRACTABLE_STRINGS:
+            json_path = parse(extractable_string["json_path"])
 
-        for key in self.keys_to_translate:
-            with_context = key in self.keys_with_context
-
-            for pointer in find_pointers_to(self.schema, key):
-                # The 'list' property is used by both 'contents' and 'when' blocks.
-                if "/when/" in pointer:
-                    continue
-
-                schema_element = resolve_pointer(self.schema, pointer)
-                if isinstance(schema_element, dict):
-                    yield from self._get_translatable_items_from_dict_element(
-                        pointer, schema_element, with_context
-                    )
-
-                elif isinstance(schema_element, list):
-                    yield from self._get_translatable_items_from_list_element(
-                        pointer, schema_element, with_context
-                    )
-
-                else:
-                    yield self._get_translatable_item(
-                        pointer, schema_element, with_context
-                    )
-
-    def _get_translatable_items_from_dict_element(
-        self, pointer, schema_element, with_context
-    ):
-        plural_forms = schema_element.get("text_plural", {}).get("forms")
-        if plural_forms:
-            plural_forms_pointer = f"{pointer}/text_plural/forms"
-            yield self._get_translatable_item(
-                plural_forms_pointer, plural_forms, with_context
-            )
-        elif "text" in schema_element:
-            placeholder_pointer = f"{pointer}/text"
-            yield self._get_translatable_item(
-                placeholder_pointer, schema_element["text"], with_context
-            )
-        else:
-            for element, value in schema_element.items():
-                element_pointer = f"{pointer}/{element}"
-                yield self._get_translatable_item(element_pointer, value, with_context)
-
-    def _get_translatable_items_from_list_element(
-        self, pointer, schema_element, with_context
-    ):
-
-        for index, element in enumerate(schema_element):
-            if isinstance(element, dict):
-                yield from self._get_translatable_items_from_dict_element(
-                    pointer, element, with_context
+            for match in json_path.find(self.schema):
+                json_pointer, string_value = self._get_json_pointer_and_string_value(
+                    str(match.full_path), match.value
```

Comment:
I'm not clear on why a json pointer is returned? Can't we migrate to using jsonpath fully?

Reply:
Ah, I think I remember - jsonpointer has the ability to update pointers within a piece of data. We should probably consider writing an extension to extend jsonpath to allow this too.

Reply:
Yeah, we use the JSON pointers to rewrite the schema when translating. I also think being able to extract the JSON pointers for all strings in a schema is useful, given that JSON pointer is a standard and JSONPath is not.

Reply:
I think we should really be looking at adopting one over the other. Especially with trying to refactor out common features such as schema navigation.

Reply:
Possibly, but as it's not possible right now I don't think it should hold this work up.